File: tests/machine/x/scala/dataset_where_filter.scala
object dataset_where_filter {
  case class Adult(name: String, age: Int, is_senior: Boolean)
  case class People(name: String, age: Int)

  val people = List(People(name = "Alice", age = 30), People(name = "Bob", age = 15), People(name = "Charlie", age = 65), People(name = "Diana", age = 45))
  val adults = for { person <- people; if person.age >= 18 } yield Adult(name = person.name, age = person.age, is_senior = person.age >= 60)
  def main(args: Array[String]): Unit = {
    println("--- Adults ---")
    for(person <- adults) {
      println(person.name + " " + "is" + " " + person.age + " " + if (person.is_senior) " (senior)" else "")
    }
  }
}

File: tests/machine/x/cs/dataset_where_filter.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main()
    {
        List<People> people = new List<People> { new People { name = "Alice", age = 30 }, new People { name = "Bob", age = 15 }, new People { name = "Charlie", age = 65 }, new People { name = "Diana", age = 45 } };
        List<Adult> adults = people.Where(person => (person.age >= 18)).Select(person => new Adult { name = person.name, age = person.age, is_senior = (person.age >= 60) }).ToList();
        Console.WriteLine("--- Adults ---");
        foreach (var person in adults)
        {
            Console.WriteLine(string.Join(" ", new[] { Convert.ToString(person.name), Convert.ToString("is"), Convert.ToString(person.age), Convert.ToString((person.is_senior ? " (senior)" : "")) }));
        }
    }
    public class People
    {
        public string name;
        public int age;
    }


    public class Adult
    {
        public string name;
        public int age;
        public bool is_senior;
    }



}
